[dingZvel] iP

dingZvel opened the pull request:
### Lee

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

**Lee** frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. Download the latest release from [here](https://github.com/dingZvel/ip/releases).
2. Copy the jar file into an empty folder. Open a terminal window in that folder.
3. Run the command 
```bash
java -jar "Lee.jar"
```
4. Add some tasks
5. let Lee manage your tasks for you 😉

And it is FREE!

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dingZvel committed:
[Level 0] Rename, Greet, Exit

dingZvel committed:
[Level-1] Echo

dingZvel committed:
[Level 2] Add, List

dingZvel committed:
[Level 3] Mark as Done

dingZvel committed:
[Level-4] ToDos, Events, Deadlines

dingZvel committed:
[A-TextUiTesting] Automated Text UI Testing

dingZvel committed:
[Level-5] Handle Errors

dingZvel committed:
[Level-6] Delete

dingZvel committed:
Add ./data/taskList.txt to load task at start up

dingZvel committed:
Save tasks list in hard disk whenever list changes

dingZvel committed:
Merge branch 'branch-Level-7'

dingZvel committed:
Change from, to and by to LocalDateTime type

dingZvel committed:
Merge branch 'branch-Level-8'

dingZvel committed:
[A-MoreOOp] Use More OOP

dingZvel committed:
[A-Packages] Organize into Packages

dingZvel committed:
Merge branch 'add-gradle-support'

dingZvel committed:
[A-Jar] Creat a JAR File

dingZvel committed:
Add Java doc

dingZvel committed:
Modify codes to follow coding standard

dingZvel committed:
Merge branch 'branch-A-CodingStandard'

sciphi-123 commented on src/main/java/lee/util/Parser.java:
```diff
@@ -0,0 +1,208 @@
+     * @param b The new status of the task.
+     * @throws LeeException If the given command syntax is incorrect.
+     */
+    private void mark(String num, boolean b) throws LeeException {
```
Would it be better to make the boolean variable name sound like a boolean instead?

sciphi-123 commented on src/main/java/lee/util/Parser.java:
```diff
@@ -0,0 +1,208 @@
+import lee.task.Deadline;
+import lee.task.Event;
+import lee.task.TaskList;
+import lee.task.ToDo;
```
Good that all classes to import are specified clearly!

sciphi-123 commented on src/main/java/lee/util/Storage.java:
```diff
@@ -0,0 +1,95 @@
+        }
+        boolean isDone = task[1].equals("1");
+        switch (task[0]) {
+        case "T" -> {
```
Good that the indentation convention for switch-case is followed!

sciphi-123 commented on src/main/java/lee/Lee.java:
```diff
@@ -0,0 +1,64 @@
+        }
+        ui.exitUi();
+        sc.close();
+    }
```
LGTM!

sciphi-123 reviewed (commented):
Code is well-organised. Good job!

arnavbajpai commented on src/main/java/lee/task/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+    private final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(" yyyy-MM-dd HH:mm");
```
Since this is the declaration of a constant maybe make use of keyword 'static'

arnavbajpai commented on src/main/java/lee/util/Parser.java:
```diff
@@ -0,0 +1,208 @@
+        String[] commands = command.split(" ");
+        String first = commands[0];
+        try {
+            if (command.equals("list")) {
```
Good that correct indentation for if else is followed 👍

arnavbajpai commented on src/main/java/lee/ui/Ui.java:
```diff
@@ -0,0 +1,49 @@
+ * Aggregates all ui related methods for tui.
+ */
+public class Ui {
+    private final String HORlINE;
```
Does this follow constant naming convention ?

arnavbajpai commented on src/main/java/lee/util/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @throws LeeException If the data file is corrupted.
+     */
+    private void addTaskFromLine(ArrayList<Task> tasks, String line) throws LeeException {
+        String[] task = line.split("\\|");
```
Maybe use plural form naming here.

dingZvel committed:
[Level-9] Find

dingZvel committed:
Merge branch 'branch-Level-9'

dingZvel committed:
Add javadoc to new methods introduced in last commit

dingZvel committed:
Refactor Parse.java and extract out all ui related methods to Ui.java

dingZvel committed:
[Level-10] GUI

dingZvel committed:
Merge branch 'branch-Level-10'

dingZvel committed:
A-Assertions: Add assertions to show assumptions

The command formats and data line format stored in data files
should not be corrupted.

Add several assertions to make sure the assumption that all formats
are correct and the data file should also be successfully created by
us if it didn't exist.

* command format is correct, even user type nothing and submit directly
* data file should be created if not existed

dingZvel committed:
Improve code quality in Storage.java

Together with commit "Refactor Parse.java..."(a45acd), improve
the code quality for better readability

* Avoid long method (in commit a45acd)
* SLAP hard (in commit a45acd)
* Make the happy path prominent
* Avoid complicated expressions

dingZvel committed:
Merge pull request #2 from dingZvel/branch-A-Assertions

A-Assertions: Add assertions to show assumptions

dingZvel committed:
Merge branch 'master' into branch-A-CodeQuality

dingZvel committed:
Merge pull request #3 from dingZvel/branch-A-CodeQuality

Improve code quality in Storage.java

dingZvel committed:
Add GUI title

dingZvel committed:
[B-Snooze] Postponing deadline task

dingZvel committed:
Add README and Ui screenshot

dingZvel committed:
[A-JUnit] Add JUnit Tests
